NeuroID/src/main/java/com/neuroid/tracker/service/NIDSamplingService.kt

```diff
@@ -0,0 +1,85 @@
+package com.neuroid.tracker.service
+
+import com.neuroid.tracker.utils.NIDLogWrapper
+import com.neuroid.tracker.utils.RandomGenerator
+
+class NIDSamplingService(
+    val logger: NIDLogWrapper,
+    // we need this to be replaceable for the tests!
+    var randomGenerator: RandomGenerator,
+    var configService: NIDRemoteConfigService,
+) {
+    private var isSessionFlowSampled = true
+
+    /**
+     * given a site id, tell me if we sample events or not. the site ID will be compared
+     */
+    fun updateIsSampledStatus(siteID: String?) {
+        val currentSampleRate = retrieveSampleRate(siteID)
+        if (currentSampleRate >= MAX_SAMPLE_RATE) {
+            logger.d(msg="updateIsSampledStatus() we are sampling at max or > than max ($currentSampleRate)")
+            isSessionFlowSampled = true
+            return
+        }
+        val randomValue = randomGenerator.getRandom(MAX_SAMPLE_RATE)
+        logger.d(msg="updateIsSampledStatus() random value: $randomValue sampleRate: $currentSampleRate")
+        if (randomValue < currentSampleRate) {
+            isSessionFlowSampled = true
+            logger.d(msg="updateIsSampledStatus() we are sampling")
+            return
+        }
+
+        logger.d(msg="updateIsSampledStatus() we are not sampling")
+        isSessionFlowSampled = false
+    }
+
+
+    /**
+     * Return the sample rate, 0 indicates sample everything, 100 indicates sample nothing!
```

Comment:
Isn't this backwards? If we are at 100 we should sample everything and if we are at 0 we sample nothing?

Reply:
coment fixed.